src/libdrakvuf/libdrakvuf.h

```diff
@@ -182,15 +182,22 @@ typedef enum memaccess_type {
     POST
 } memaccess_type_t;
 
+typedef struct process_data {
+    const char *name;   /* Process name */
+    vmi_pid_t pid ;     /* Process pid */
+    vmi_pid_t ppid ;    /* Process parent pid */
+    addr_t base_addr ;  /* Process base address */
+    int64_t userid ;    /* Process SessionID/UID */
+} proc_data_t ;
+
 typedef struct drakvuf* drakvuf_t;
 struct drakvuf_trap;
 typedef struct drakvuf_trap drakvuf_trap_t;
 
 typedef struct drakvuf_trap_info {
     unsigned int vcpu;
     uint16_t altp2m_idx;
-    const char* procname; /* Currently executing process' name */
-    int64_t userid; /* Currently executing process' SessionID/UID */
+    proc_data_t *proc_data ; /* Current executing process data */
```

Comment:
So one issue I have with this is that this will has to be malloced/freed for every trap which adds to the overhead quite a bit. I think we should explore the option of having this as an embedded structure within the trap info. Then for `linux_get_current_process_data` we could pass it in as an optional input so that it can forgo the malloc step.

Reply:
It makes sense, I'm going to take a look